src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resultStudents: Student[] = JSON.parse(JSON.stringify(students));
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const averageMarks = (grades: number[]): number => grades
+    .reduce((a, b) => a + b, 0) / grades.length;
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? resultStudents.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
+        : resultStudents.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
+    case SortType.Age:
+    case SortType.Married:
+      return order === 'asc'
+        ? resultStudents.sort((a, b) => +a[sortBy] - +b[sortBy])
+        : resultStudents.sort((a, b) => +b[sortBy] - +a[sortBy]);
+
+    case SortType.AverageGrade:
+      return order === 'asc'
+        ? resultStudents
+          .sort((a, b) => averageMarks(a[sortBy]) - averageMarks(b[sortBy]))
+        : resultStudents
+          .sort((a, b) => averageMarks(b[sortBy]) - averageMarks(a[sortBy]));
```

Comment:
- in this case you use sort just once
- add empty lines between cases

```suggestion
return [...students].sort((a, b) => {
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname: order === 'asc'
        ?  a[sortBy].localeCompare(b[sortBy]))
        : b[sortBy].localeCompare(a[sortBy]));
        
    case SortType.Age:
    case SortType.Married:
      return order === 'asc'
        ?  +a[sortBy] - +b[sortBy])
        :  +b[sortBy] - +a[sortBy]);

    case SortType.AverageGrade:
      return order === 'asc'
        ?  averageMarks(a[sortBy]) - averageMarks(b[sortBy]))
        : averageMarks(b[sortBy]) - averageMarks(a[sortBy]));
```